kernel/src/sched.rs

```diff
@@ -902,37 +903,101 @@ impl Kernel {
                 upcall_ptr,
                 appdata,
             } => {
```

Comment:
@phil-levis Can you read through the subscribe match block, and check that the logic flow and comments match what TRD104 promises?

Reply:
Will do.